pkg/component/data/pinecone/v0/io.go

```diff
@@ -1,28 +1,23 @@
-// TASK_QUERY and TASK_RERANK are not refactored yet, they will be addressed in ins-7102
+// TODO: TASK_QUERY and TASK_RERANK are not refactored yet, they will be
+// addressed in INS-7102.
 package pinecone
 
+type vector struct {
+	ID       string            `json:"id" instill:"id"`
+	Values   []float64         `json:"values,omitempty" instill:"values"`
+	Metadata map[string]string `json:"metadata,omitempty" instill:"metadata"`
```

Comment:
TODO: we need to make this `map[string]any` but `ReadData` fails (cc @donch1989)